addon/mixins/dom.js

```diff
@@ -133,29 +133,33 @@ export default Mixin.create({
     }
 
     // We cannot use Array.findIndex as we cannot rely on babel/polyfill being present
-    for (let i = 0; i < this._listeners.length; i++) {
-      let listener = this._listeners[i];
+    for (let i = 0; i < this._listeners.length; i += 5) {
       if (
-        listener.element === element &&
-        listener.eventName === eventName &&
-        listener._callback === callback
+        this._listeners[i] === element &&
+        this._listeners[i + 1] === eventName &&
+        this._listeners[i + 3] === callback
       ) {
         /*
          * Drop the event listener and remove the listener object
          */
-        element.removeEventListener(eventName, listener.callback, options);
-        this._listeners.splice(i, 1);
+        let ownCallback = this._listeners[i + 2];
+        element.removeEventListener(eventName, ownCallback, options);
+        this._listeners.splice(i, 5);
         break;
       }
     }
   },
 
   willDestroy() {
     this._super(...arguments);
-    if (Array.isArray(this._listeners)) {
+    if (this._listeners !== undefined) {
       /* Drop non-passive event listeners */
-      for (let i = 0; i < this._listeners.length; i++) {
-        let { element, eventName, callback, options } = this._listeners[i];
+      for (let i = 0; i < this._listeners.length; i += 5) {
+        let element = this._listeners[i];
```

Comment:
Lets make these use a constant to make it super clear what we are doing, I'm thinking something like:

```js
let element = this._listeners[i + ELEMENT_INDEX];
let eventName = this._listeners[i + EVENT_NAME_INDEX];
let callback = this._listeners[i + CALLBACK_INDEX];
let options = this._listeners[i + OPTIONS_INDEX];
```

What do you think?

Reply:
yep, nice idea will update PR and will make same in bb